lib/pharos/config.rb

```diff
@@ -33,6 +33,29 @@ def self.load(raw_data)
       config = new(schema_data)
       config.data = raw_data.freeze
 
+      config.hosts.reject { |h| h.bastion.nil? }.each do |host_a|
+        host_b = config.hosts.find { |host| host_a.bastion.attributes.all? { |k, v| host.attributes[k] == v } }
```

Comment:
Quite complex block, maybe could be refactored to a separate method?

Reply:
I used to have the `def ==(other)` in bastion but i think it looked more confusing:`host_a.bastion == host`

Maybe create some `attributes_equal?` instead of overriding `==`